File: Frontend/students-employment-records-site/src/domain/students/enums/studentStatus.ts
import { enumToArrayNumber } from "@/tools/enums/enumUtils"
import { NeverUnreachable } from "@/tools/neverUreachable"

export enum StudentStatus {
    Active = 1,
    Expelled = 2,
    Finished = 3,
}

export namespace StudentStatus {
    export function getAll(): StudentStatus[] {
        return enumToArrayNumber<StudentStatus>(StudentStatus)
    }

    export function getDisplayText(status: StudentStatus): string {
        switch (status) {
            case StudentStatus.Active:
                return "Обучается"
            case StudentStatus.Expelled:
                return "Отчислен"
            case StudentStatus.Finished:
                return "Отчислен, в связи с окончанием обучения"
            default:
                throw new NeverUnreachable(status)
        }
    }
}

File: Backend/SER.Domain/Students/Enums/StudentStatus.cs
namespace SER.Domain.Students.Enums;
public enum StudentStatus
{
	Active = 1,
	Expelled = 2,
	Finished = 3,
	AcademicLeave = 4
}

public static class StudentStatusExtensions
{
	public static String DisplayName(this StudentStatus status)
	{
		return status switch
		{
			StudentStatus.Active => "Обучается",
			StudentStatus.Expelled => "Отчислен",
			StudentStatus.Finished => "Отчислен, в связи с окончанием обучения",
			StudentStatus.AcademicLeave => "Академический отпуск",
			_ => throw new Exception()
		};
	}
}
